torch/csrc/api/include/torch/data/datasets/chunk.h

```diff
@@ -332,6 +367,15 @@ class ChunkDataset final
       "The requested batch size does not match with the initialized batch size.\n"
       " The requested batch size is ", batch_size,
       ", while the dataset is created with batch size equal to ", options_.batch_size_);
+    
+    batch_inteval_count_ ++;
```

Comment:
get_batch can be executed concurrently, so we need to move the lock outside right?

Reply:
Need to see how this works more, we don't need to have a synchronization on each call. Let's discuss more.

Reply:
get_batch() actually is under a lock. When using multiple workers, the data is pushed into a torch::data::detailed::queue. Every operation (push, pop, clear) of the queue is under a mutex. So we are good here :-)

Reply:
Let's see if we can avoid this as we discussed offline.